hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECBlockOutputStreamEntry.java

```diff
@@ -384,6 +388,33 @@ private Stream<ECBlockOutputStream> dataStreams() {
         .filter(Objects::nonNull);
   }
 
+  public ByteString calculateChecksum() throws IOException {
+    if (blockOutputStreams == null) {
+      throw new IOException("Block Output Stream is null");
+    }
+
+    List<ContainerProtos.ChunkInfo> chunkInfos = new ArrayList<>();
+    // First chunk should always have the additional chunks in a partial stripe.
+    int currentIdx = blockOutputStreams[0]
+        .getContainerBlockData().getChunksCount();
+    for (ECBlockOutputStream stream: blockOutputStreams) {
+      List<ContainerProtos.ChunkInfo> chunks =
+          stream.getContainerBlockData().getChunksList();
+      if (chunks.size() > currentIdx - 1) {
+        chunkInfos.add(chunks.get(currentIdx - 1));
```

Comment:
We should be able to eliminate the temporary array list chunkInfos, for sake of performance. As a future item, we should do some kind of performance benchmarks to understand the impact of file checksum.

Reply:
I added it to make the code readable but I guess it can be removed.